internal/handshake/params_negotiator.go

```diff
@@ -0,0 +1,104 @@
+package handshake
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/lucas-clemente/quic-go/internal/protocol"
+	"github.com/lucas-clemente/quic-go/internal/utils"
+)
+
+type paramsNegotiator struct {
+	paramsNegotiatorBase
+}
+
+var _ ParamsNegotiator = &paramsNegotiator{}
+
+// newParamsNegotiator creates a new connection parameters manager
+func newParamsNegotiator(pers protocol.Perspective, v protocol.VersionNumber, params *TransportParameters) *paramsNegotiator {
+	h := &paramsNegotiator{}
+	h.perspective = pers
+	h.version = v
+	h.init(params)
+	return h
+}
+
+func (h *paramsNegotiator) SetFromTransportParameters(params []transportParameter) error {
+	var foundInitialMaxStreamData bool
+	var foundInitialMaxData bool
+	var foundInitialMaxStreamID bool
+	var foundIdleTimeout bool
+
+	for _, p := range params {
+		switch p.Parameter {
+		case initialMaxStreamDataParameterID:
+			foundInitialMaxStreamData = true
+			if len(p.Value) != 4 {
+				return fmt.Errorf("wrong length for initial_max_stream_data: %d", len(p.Value))
```

Comment:
Should we include the expected len in the error message?

Reply:
We can do that.